src/tools/compiletest/src/runtest.rs

```diff
@@ -428,7 +434,12 @@ impl<'test> TestCx<'test> {
     }
 
     fn check_correct_failure_status(&self, proc_res: &ProcRes) {
-        let expected_status = Some(self.props.failure_status);
+        let expected_status =
+            if self.props.should_ice {
+                Some(101)
+            } else {
+                Some(self.props.failure_status)
+            };
```

Comment:
Would you mind moving this adaptation of the failure status to `header.rs` where we already do something similar:
https://github.com/rust-lang/rust/blob/e4931eaaa3d95189b30e90d3af9f0db17c41bbb0/src/tools/compiletest/src/header.rs#L574-L579

Then it's all in one place.